src/datatypes/ip.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2012-2015 Metamarkets Group Inc.
+ * Copyright 2015-2020 Imply Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Class, Instance } from 'immutable-class';
+import { Address4, Address6 } from 'ip-address';
+
+import { PlyType } from '../types';
+// TODO: Goes in plywood.js after compile. have to figure out why this disappears after compile
+// const { Address4, Address6 } = require('ip-address');
```

Comment:
@jgoz can you help me resolve this please?

Reply:
You have to add this to `extra/prefix.js`:

```js
var ipAddress = require('ip-address');
var Address4 = ipAddress.Address4;
var Address6 = ipAddress.Address6;
```

Reply:
So based on some issues I was having with module resolution for `is-ip: 5.0.0` I had to downgrade to `3.1.0`.

Reply:
Can you elaborate on the issues? Using any version other than 5.0 will result in multiple copies of the same dependency+tree being included in application bundles so it would be preferable to be in sync.

Reply:
It was related to module resolution (see error below). After plywood is expecting commonjs modules after compiling and `is-ip` uses ECMAScript. I tried a few different approaches but the easiest solution came down to just using an older version. I understand the tradeoff is two separate versions in the application bundle but the package is very small so didn't see it as a concern.

```
internal/modules/cjs/loader.js:1080
      throw new ERR_REQUIRE_ESM(filename, parentPath, packageJsonPath);
      ^

Error [ERR_REQUIRE_ESM]: Must use import to load ES Module: /Users/tylerreece/Development/plywood/node_modules/is-ip/index.js
require() of ES modules is not supported.
require() of /Users/tylerreece/Development/plywood/node_modules/is-ip/index.js from /Users/tylerreece/Development/plywood/build/plywood.js is an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which defines all .js files in that package scope as ES modules.
Instead rename index.js to end in .cjs, change the requiring code to use import(), or remove "type": "module" from /Users/tylerreece/Development/plywood/node_modules/is-ip/package.json.
```

Reply:
Thanks for the explanation, I figured it was something like this. The author of `is-ip` is an ESM fundamentalist, so I should have predicted this 😄

Reply:
😆  no problem.